packages/web-ui-registration/src/LoginForm.tsx

```diff
@@ -0,0 +1,179 @@
+import { useUniqueId } from '@rocket.chat/fuselage-hooks';
+import { FieldGroup, TextInput, Field, PasswordInput, ButtonGroup, Button, Callout } from '@rocket.chat/fuselage';
+import { Form, ActionLink } from '@rocket.chat/layout';
+import { useLoginWithPassword, useSetting } from '@rocket.chat/ui-contexts';
+import type { ReactElement } from 'react';
+import { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { Trans, useTranslation } from 'react-i18next';
+
+import EmailConfirmationForm from './EmailConfirmationForm';
+import type { DispatchLoginRouter } from './hooks/useLoginRouter';
+import Services from './Services';
+
+type LoginErrors =
+	| 'error-user-is-not-activated'
+	| 'error-invalid-email'
+	| 'error-login-blocked-for-ip'
+	| 'error-login-blocked-for-user'
+	| 'error-license-user-limit-reached'
+	| 'user-not-found'
+	| 'error-app-user-is-not-allowed-to-login';
+
+export const LoginForm = ({ setLoginRoute }: { setLoginRoute: DispatchLoginRouter }): ReactElement => {
+	const {
+		register,
+		handleSubmit,
+		setError,
+		clearErrors,
+		getValues,
+		formState: { errors },
+	} = useForm<{
+		email?: string;
+		username: string;
+		password: string;
+	}>({
+		mode: 'onChange',
+	});
+
+	const formLabelId = useUniqueId();
+
+	const [errorOnSubmit, setErrorOnSubmit] = useState<LoginErrors | undefined>(undefined);
```

Comment:
We don't need to `setState` for the errors, we can send validations to the `register` and handle multiple errors at once